oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/web/UserConsentPage.java

```diff
@@ -0,0 +1,116 @@
+package org.springframework.security.oauth2.server.authorization.web;
+
+import org.springframework.http.MediaType;
+import org.springframework.security.oauth2.core.endpoint.OAuth2AuthorizationRequest;
+import org.springframework.security.oauth2.core.endpoint.OAuth2ParameterNames;
+import org.springframework.security.oauth2.core.oidc.OidcScopes;
+import org.springframework.security.oauth2.server.authorization.OAuth2Authorization;
+import org.springframework.security.oauth2.server.authorization.client.RegisteredClient;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * User Consent Page
+ *
+ * @author shanhy
+ * @date 2021/3/30 15:42
+ */
+public interface UserConsentPage {
+
+	MediaType TEXT_HTML_UTF8 = new MediaType("text", "html", StandardCharsets.UTF_8);
+	String CONSENT_ACTION_PARAMETER_NAME = "consent_action";
+	String CONSENT_ACTION_APPROVE = "approve";
+	String CONSENT_ACTION_CANCEL = "cancel";
+
+	default void displayConsent(HttpServletRequest request, HttpServletResponse response,
```

Comment:
remove UserConsentPage interface default method , enable util class or impl class 。

Reply:
OK，Implementation of DefaultGenerateUserConsentPage

Reply:
1. isConsentApproved() and isConsentCancelled() retain default.
2. displayConsent() and generateConsentPage() impl class.